File: 3823-reverse-letters-then-special-characters-in-a-string/3823-reverse-letters-then-special-characters-in-a-string.cpp
class Solution {
public:
    string reverseByType(string s) {
        string lower = "";
        string charector = "";
        int i = 0;
        int n = s.size();
        while(i < n){
            if(isalpha(s[i])){
                lower += s[i];
            }else{
                charector += s[i];
            }
            i++;
        }
        reverse(lower.begin(), lower.end());
        reverse(charector.begin(), charector.end());

        string ans ="";
        int l = 0;
        int c = 0;

        for(int i = 0; i < n; i++){
            if(isalpha(s[i])){
                ans += lower[l++];
            }else{
                ans += charector[c++];
            }
        }
        return ans;
    }
};

File: 3823-reverse-letters-then-special-characters-in-a-string/3823-reverse-letters-then-special-characters-in-a-string.java
class Solution {
    public String reverseByType(String s) {
        String letter = "";
        String symbol = "";

        for(char ch : s.toCharArray()){
            if(Character.isLetter(ch)){
                letter += ch;
            }else{
                symbol += ch;
            }
        }
        String ans = "";
        int i = letter.length() - 1;
        int j = symbol.length() - 1;

        for(char ch : s.toCharArray()){
            if(Character.isLetter(ch)){
                ans += letter.charAt(i);
                i--;
            }else{
                ans += symbol.charAt(j);
                j--;
            }
        }
        return ans;
    }
}